streaming/src/main/scala/org/apache/spark/streaming/rdd/WriteAheadLogBackedBlockRDD.scala

```diff
@@ -135,8 +135,11 @@ class WriteAheadLogBackedBlockRDD[T: ClassTag](
         // FileBasedWriteAheadLog will not create any file or directory at that path. Also,
         // this dummy directory should not already exist otherwise the WAL will try to recover
         // past events from the directory and throw errors.
+        // Specifically, the nonExistentDirectory will contain a colon in windows, this is invalid
+        // for hadoop. Remove the drive letter and colon, e.g. "D:" out of this path by default
         val nonExistentDirectory = new File(
-          System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString).getAbsolutePath
+          System.getProperty("java.io.tmpdir").replaceFirst("[a-zA-Z]:", ""),
+          UUID.randomUUID().toString).getPath
```

Comment:
I think this is the same, if you prefer this change, I could add another commit.